module/zfs/trace.c

```diff
@@ -41,10 +42,14 @@
 #include <sys/trace.h>
 #include <sys/trace_acl.h>
 #include <sys/trace_arc.h>
+#if !defined(HAVE_DECLARE_EVENT_CLASS)
+#include <sys/trace_dbgmsg.h>
+#endif
```

Comment:
trace_dbgmsg.h appears to handle `!defined(HAVE_DECLARE_EVENT_CLASS)`; why do we need to omit it in that case?

Reply:
I was working around the trace_dbgmsg.h relying on DEFINE_DTRACE_PROBE1, but that can be handled better in include/sys/trace.h by including trace_dbgmsg.h there on the non-tracepoint side after DEFINE_DTRACE_PROBE1 is defined.